core/cat/looking_glass/agent_manager.py

```diff
@@ -145,26 +143,53 @@ async def execute_form_agent(self, stray):
             else:
                 # continue form
                 return active_form.next()
-        
-        return None # no active form
-        
-    async def execute_memory_chain(self, agent_input, prompt_prefix, prompt_suffix, stray):
-
-        input_variables = [i for i in agent_input.keys() if i in prompt_prefix + prompt_suffix]
-        # memory chain (second step)
-        memory_prompt = PromptTemplate(
-            template = prompt_prefix + prompt_suffix,
-            input_variables=input_variables
+
+        return None  # no active form
+
+    async def execute_memory_chain(
+        self, agent_input, prompt_prefix, prompt_suffix, stray
+    ):
+        input_variables = {i: j for i, j in agent_input.items() if i in prompt_prefix}
+
+        chat_history = []
+        for message in stray.working_memory.history:
```

Comment:
This could be a method in `WorkingMemory` because convo history always stays there.
Also as a facility we can think about `cat.llm(prompt, history=True)` to include chat history into direct LLM calls 
Maybe `cat.llm(prompt, history=custom_history)` to override the one in working memory